### ISSUE TITLE
no-error-pmd execution is failing if version in not changed to 10.4

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#12129]
right after release 10.3.2 I forgot to update back to 10.4-SNAPSHOT

resulted in build failure:
https://app.circleci.com/pipelines/github/checkstyle/checkstyle/14472/workflows/e0cd3819-c488-4b41-af26-d5debdf521a1/jobs/164841

it became green right after bump
https://app.circleci.com/pipelines/github/checkstyle/checkstyle/14474/workflows/873a3d0a-5015-458c-a481-c651efb3cd09/jobs/164876

problems is that pmd takes this SNAPSHOTS from maven central, it is not what we need!!! :
```
~/java/github/romani/checkstyle/.ci-temp/pmd [master|✔] 
$ ./mvnw verify --show-version --errors --batch-mode                 -DskipTests                 -Dmaven.javadoc.skip=true                 -Dmaven.source.skip=true                 -Dpmd.skip=true                 -Dcheckstyle.skip=false                 -Dcheckstyle.version="10.3.4-SNAPSHOT"
Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
Maven home: /home/rivanov/.m2/wrapper/dists/apache-maven-3.8.6-bin/1ks0nkde5v1pk9vtc31i9d0lcd/apache-maven-3.8.6
Java version: 11.0.14.1, vendor: Amazon.com Inc., runtime: /opt/jvm/amazon-corretto-11.0.14.10.1-linux-x64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.4.0-210-generic", arch: "amd64", family: "unix"
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 39 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 39 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] PMD                                                                [pom]
[INFO] PMD Core                                                           [jar]
[INFO] PMD Test Schema                                                    [jar]
[INFO] PMD Test Framework                                                 [jar]
[INFO] PMD language module testing utilities                              [jar]
[INFO] PMD C++                                                            [jar]
[INFO] PMD C#                                                             [jar]
[INFO] PMD Dart                                                           [jar]
[INFO] PMD Fortran                                                        [jar]
[INFO] PMD Gherkin                                                        [jar]
[INFO] PMD Go                                                             [jar]
[INFO] PMD Groovy                                                         [jar]
[INFO] PMD Lua                                                            [jar]
[INFO] PMD Java                                                           [jar]
[INFO] PMD JSP                                                            [jar]
[INFO] PMD Kotlin                                                         [jar]
[INFO] PMD Matlab                                                         [jar]
[INFO] PMD Modelica                                                       [jar]
[INFO] PMD Perl                                                           [jar]
[INFO] PMD Objective-C                                                    [jar]
[INFO] PMD PHP                                                            [jar]
[INFO] PMD PL/SQL                                                         [jar]
[INFO] PMD Python                                                         [jar]
[INFO] PMD Ruby                                                           [jar]
[INFO] PMD Swift                                                          [jar]
[INFO] PMD Velocity                                                       [jar]
[INFO] PMD XML and XSL                                                    [jar]
[INFO] PMD Apex Jorje Parser Library                                      [pom]
[INFO] PMD Apex                                                           [jar]
[INFO] PMD HTML                                                           [jar]
[INFO] PMD JavaScript                                                     [jar]
[INFO] PMD Scala Common Source and Settings                               [pom]
[INFO] PMD Scala for Scala 2.13                                           [jar]
[INFO] PMD Visualforce                                                    [jar]
[INFO] PMD Distribution Packages                                          [jar]
[INFO] PMD Java 8 Integration                                             [jar]
[INFO] PMD Documentation Generator                                        [jar]
[INFO] PMD Scala - Transitional package (deprecated)                      [pom]
[INFO] PMD Scala for Scala 2.12                                           [jar]
[INFO] 
[INFO] ----------------------< net.sourceforge.pmd:pmd >-----------------------
[INFO] Building PMD 6.49.0-SNAPSHOT                                      [1/39]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (enforce-versions) @ pmd ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (enforce-no-snapshots) @ pmd ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (enforce-plugin-versions) @ pmd ---
[INFO] 
[INFO] --- maven-javadoc-plugin:3.2.0:jar (attach-javadocs) @ pmd ---
[INFO] Skipping javadoc generation
[INFO] 
[INFO] >>> maven-source-plugin:3.2.1:jar (attach-sources) > generate-sources @ pmd >>>
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (enforce-versions) @ pmd ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (enforce-no-snapshots) @ pmd ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (enforce-plugin-versions) @ pmd ---
[INFO] 
[INFO] <<< maven-source-plugin:3.2.1:jar (attach-sources) < generate-sources @ pmd <<<
[INFO] 
[INFO] 
[INFO] --- maven-source-plugin:3.2.1:jar (attach-sources) @ pmd ---
[INFO] Skipping source per configuration.
[INFO] 
[INFO] >>> maven-pmd-plugin:3.18.0:check (default) > :pmd @ pmd >>>
[INFO] 
[INFO] --- maven-pmd-plugin:3.18.0:pmd (pmd) @ pmd ---
[INFO] Skipping PMD execution
[INFO] 
[INFO] <<< maven-pmd-plugin:3.18.0:check (default) < :pmd @ pmd <<<
[INFO] 
[INFO] 
[INFO] --- maven-pmd-plugin:3.18.0:check (default) @ pmd ---
[INFO] Skipping PMD execution
[INFO] 
[INFO] >>> maven-pmd-plugin:3.18.0:cpd-check (default) > :cpd @ pmd >>>
[INFO] 
[INFO] --- maven-pmd-plugin:3.18.0:cpd (cpd) @ pmd ---
[INFO] 
[INFO] <<< maven-pmd-plugin:3.18.0:cpd-check (default) < :cpd @ pmd <<<
[INFO] 
[INFO] 
[INFO] --- maven-pmd-plugin:3.18.0:cpd-check (default) @ pmd ---
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.1.2:check (checkstyle-check) @ pmd ---
[INFO] Downloading from apache.snapshots: https://repository.apache.org/snapshots/com/puppycrawl/tools/checkstyle/10.3.4-SNAPSHOT/maven-metadata.xml
[INFO] Downloading from sonatype-nexus-plugin-snapshots: https://oss.sonatype.org/content/repositories/snapshots/com/puppycrawl/tools/checkstyle/10.3.4-SNAPSHOT/maven-metadata.xml
[INFO] Downloaded from sonatype-nexus-plugin-snapshots: https://oss.sonatype.org/content/repositories/snapshots/com/puppycrawl/tools/checkstyle/10.3.4-SNAPSHOT/maven-metadata.xml (1.4 kB at 1.5 kB/s)
[INFO] Downloading from sonatype-nexus-plugin-snapshots: https://oss.sonatype.org/content/repositories/snapshots/com/puppycrawl/tools/checkstyle/10.3.4-SNAPSHOT/checkstyle-10.3.4-20220828.165710-2.pom

```

**Comment by Rahulkhinchi03** [comment]
@romani here you mean that the PMD needs the maven version but without -SNAPSHOT?

**Comment by nrmancuso** [comment]
Closed via #12130
